### PR TITLE
Initial merge

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Description

Not much change to the module but added automation to validate the terraform with static analysis, secret detection, `README.md` generation, `module-metadata.json` generation, removed unused variables, fixed formating, added renovate to update dependencies and our common Terratest tests. 

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [x] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: module.yaml [hunk 1]
```diff
@@ -35,6 +34,3 @@ versions:
       scope: ignore
     - name: ibmcloud_api_key
       scope: global
-    - name: common_tags
```
Comment (triceam) [inline module.yaml:38]: why were common tags removed?  this was explicitly added in the last commit

File: module.yaml [hunk 2]
```diff
@@ -22,7 +21,7 @@ versions:
   dependencies:
     - id: resource_group
       refs:
-        - source: github.com/terraform-ibm-modules/terraform-ibm-toolkit-resource-group
+        - source: github.com/cloud-native-toolkit/terraform-ibm-resource-group
```
Comment (triceam) [inline module.yaml:24]: why was this changed to the old git repo url?

**Review by triceam — COMMENTED** [review]
Overall, looks like no major functional changes.  I marked two questions about module metadata.   

Will the content in the `examples` folder be executed in addition to `test/stages`, or is it intended to replace `test/stages`?

**Comment by vburckhardt** [comment]
@triceam - I think we should continue to execute test/stages, in addition to the examples. At least for now, until we gain more confidence that nothing is breaking.

File: .github/settings.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  # Uncomment this name property and set the name to the current repo name.
+  name: "terraform-ibm-toolkit-object-storage"
+
+  # The description is displayed under the repository name on the
+  # organization page and in the 'About' section of the repository.
+
+  # Uncomment this description property
+  # and update the description to the current repo description.
+  description: "Terraform module to provision Object Storage in the IBM Cloud account"
```
Comment (vburckhardt) [inline .github/settings.yml:25]: -> Provisions an IBM Cloud Object Storage instance

This aligns better with the conventions in term of description. Interested in @SirSpidey perspective as well.

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[submodule "common-dev-assets"]
+	path = common-dev-assets
+	url = https://github.com/terraform-ibm-modules/common-dev-assets.git
```
Comment (vburckhardt) [inline .gitmodules:3]: add `branch = main` - see https://github.com/terraform-ibm-modules/terraform-ibm-module-template/blob/main/.gitmodules .

File: README.md [hunk 5]
```diff
@@ -4,23 +4,8 @@ Module to work with an IBM Cloud Object Storage instance. If the `provision` fla
 
 **Note:** This module follows the Terraform conventions regarding how provider configuration is defined within the Terraform template and passed into the module - https://www.terraform.io/docs/language/modules/develop/providers.html. The default provider configuration flows through to the module. If different configuration is required for a module, it can be explicitly passed in the `providers` block of the module - https://www.terraform.io/docs/language/modules/develop/providers.html#passing-providers-explicitly.
```
Comment (vburckhardt) [inline README.md:5]: Is it still needed? These conventions are pretty much mainstream at this stage, so it may be safe to assume consumer is aware of those?

File: README.md [hunk 6]
```diff
@@ -4,23 +4,8 @@ Module to work with an IBM Cloud Object Storage instance. If the `provision` fla
...
-### Terraform providers
-
-- IBM Cloud provider >= 1.18
-
-## Module dependencies
-
-- Resource group - github.com/terraform-ibm-modules/terraform-ibm-toolkit-resource-group
-
-## Example usage
+## Usage
```
Comment (vburckhardt) [inline README.md:8]: Could we align the usage section with general guidance of using actual values (albeit dummy)

File: examples/complete/main.tf [hunk 7]
```diff
@@ -0,0 +1,10 @@
+
+module "cos_example" {
```
Comment (vburckhardt) [inline examples/complete/main.tf:2]: One suggestion here is to name it "cos_instance" (or "cos_instance_example") to make it crystal clear it is creating an instance as opposed to a bucket.

File: main.tf [hunk 8]
```diff
@@ -55,28 +54,28 @@ resource "ibm_resource_key" "cos_credentials" {
...
-  source = "cloud-native-toolkit/clis/util"
-  version = "1.16.4"
+  source  = "cloud-native-toolkit/clis/util"
+  version = "1.18.1"
 
   clis = ["jq"]
 }
 
-data external instance {
+data "external" "instance" {
```
Comment (vburckhardt) [inline main.tf:71]: @triceam @seansund - Is this script still needed. What was the context for its addition? I assume it's trying to work some limitation in the provider? We've not seen issue with waiting for the cos instance so far in PaaS, so wondering whether this is still valid.

File: main.tf [hunk 9]
```diff
@@ -55,28 +54,28 @@ resource "ibm_resource_key" "cos_credentials" {
...
 
-  //User can increase timeouts
+  # User can increase timeouts
   timeouts {
     create = "15m"
     delete = "15m"
   }
 }
 
 module "clis" {
```
Comment (vburckhardt) [inline main.tf:64]: It this really needed as opposed to indicate the dependencies to consumer (or providing a link on how to set up dependencies via a script).
There may be a few issues:
- Forcing download from external sources may be an issue for some customers running their terraform in limited connectivity / airgapped. For instance from a Schematics remote worker (Agent)
- Some customers may want to finely control the version of those dependencies - eg: to ensure use of dependency versions with no CVEs

**Review by vburckhardt — COMMENTED** [review]
Some comments

File: module.yaml (hunk 2 above)
Comment (vburckhardt) [reply module.yaml:24]: Looks like this was picked up from an older commit so unlikely that it is done on purpose. We can review with @daniel-butler-irl when he's back.

File: .gitmodules (hunk 4 above)
Comment (ocofaigh) [reply .gitmodules:3]: by default it picks primary branch, so this is `main` by default

Comment (vburckhardt) [reply .gitmodules:3]: @ocofaigh - I've seen issues locally with git submodule update until the branch was explicit, so I don't think it hurts adding it.
